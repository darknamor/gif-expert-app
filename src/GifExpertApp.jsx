import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch', 'Dragonball']);
  const onAddCategory = (e) => {
    if (categories.includes(e)) return;
    setCategories([e, ...categories]);
  };
  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory onAddCategory={onAddCategory} />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
