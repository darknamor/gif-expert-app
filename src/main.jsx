import React from 'react';
import ReactDOM from 'react-dom/client';
import { GifExpertApp } from './Components/GifExpertApp';
import './Styles/styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifExpertApp />
  </React.StrictMode>
);
