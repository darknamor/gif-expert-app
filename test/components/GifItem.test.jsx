import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components';

describe('Pruebas en <GifItem />', () => {
  const title = 'Test-title';
  const url = 'https://www.test-url.com';
  test('debe hacer match con el snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('debe demostrar la imagen con el URL y el ALT indicado', () => {
    render(<GifItem title={title} url={url} />);
    const { alt, src } = screen.getByRole('img');
    expect(src).toBe(url + '/');
    expect(alt).toBe(title);
  });

  test('debe demostrar el titulo en el componente', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
