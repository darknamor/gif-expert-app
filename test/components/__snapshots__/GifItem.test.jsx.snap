// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test-title"
      src="https://www.test-url.com"
    />
    <p>
      Test-title
    </p>
  </div>
</div>
`;
